### PR TITLE
Merge benhirsch24 work on od

### PR DESCRIPTION
**Pull request by keunwoo**
This PR just merges the work-in-progress by @benhirsch24 on issue uutils/coreutils#237 and modernizes it (so that, for example, it builds with current Rust nightly without warnings). It is very preliminary work, but I think it is best to land the WIP on master so that other people can pick it up and work on it if they like.

### CODE REVIEW

**Comment by Arcterus** [comment]
Would you mind fixing some formatting stuff?

**Comment by keunwoo** [comment]
It's not obvious to me what formatting is wrong. As noted above, most of the code is basically from @benhirsch24 . Could you give me some pointers or examples where it's wrong? I've read the Rust guidelines ( http://aturon.github.io/ ).

File: src/od/od.rs [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    let bytes = &mut [b'\x00'; 16];
+    loop {
+        match f.read(bytes) {
+            Ok(n) => {
+                print!("{:07o}", addr);
+                match radix {
+                    Radix::Decimal => {
+                    },
+                    Radix::Octal => {
+                        for b in range(0, n/std::u16::BYTES) {
```
Comment (Arcterus) [inline src/od/od.rs]: I'd prefer that there be a space between `n` and `std::u16::BYTES`.

File: src/od/od.rs [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+                            let bs = &bytes[2*b .. 2*b+2];
```
Comment (Arcterus) [inline src/od/od.rs]: I'd also like spaces between the `2` and `*`, `*` and `b`, etc.

File: src/od/od.rs [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+fn parse_radix(radix_str: Option<String>) -> Radix {
+    let rad = match radix_str {
+        Some(s) => {
+            let st = s.into_bytes();
+            if st.len() != 1 {
+                panic!("Radix must be one of [d, o, b, x]\n");
+            }
+
+            let radix: char = *(st.get(0)
+                                .expect("byte string of length 1 lacks a 0th elem")) as char;
```
Comment (Arcterus) [inline src/od/od.rs]: I also think this would be a little nicer if the `.expect` was lined up with the `.get` above.

File: src/od/od.rs [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+                        "BYTES"),
+        getopts::optopt("t", "format", "select output format or formats", "TYPE"),
+        getopts::optflag("v", "output-duplicates", "do not use * to mark line suppression"),
+        getopts::optopt("w", "width",
+                        ("output BYTES bytes per output line. 32 is implied when BYTES is not \
+                          specified."),
+                        "BYTES"),
+        getopts::optflag("h", "help", "display this help and exit."),
+        getopts::optflag("v", "version", "output version information and exit."),
+        ];
```
Comment (Arcterus) [inline src/od/od.rs]: Also, there is an extra indent here.

File: src/od/od.rs [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+    let mut addr = 0;
+    let bytes = &mut [b'\x00'; 16];
+    loop {
+        match f.read(bytes) {
+            Ok(n) => {
+                print!("{:07o}", addr);
+                match radix {
+                    Radix::Decimal => {
+                    },
```
Comment (Arcterus) [inline src/od/od.rs]: Also, I'd like the `}` to be on the same line as the `{`.

File: src/od/od.rs [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+                    Radix::Octal => {
+                        for b in range(0, n/std::u16::BYTES) {
+                            let bs = &bytes[2*b .. 2*b+2];
+                            let p: u16 = (bs[1] as u16) << 8 | bs[0] as u16;
+                            print!(" {:06o}", p);
+                        }
+                        if n % std::u16::BYTES == 1 {
+                            print!(" {:06o}", bytes[n-1]);
```
Comment (Arcterus) [inline src/od/od.rs]: Spaces between `n` and `-` and `-` and `1`.

File: README.md [hunk 7]
```diff
@@ -146,7 +146,6 @@ To do
 - mktemp
 - mv (almost done, one more option)
 - numfmt
-- od
```
Comment (Arcterus) [inline README.md]: Instead of deleting this, can you add "partially finished" or something to that effect?

**Comment by Arcterus** [comment]
These are all really nitpicky.  You haven't actually done anything wrong.

File: README.md (hunk 7 above)
Comment (keunwoo) [inline README.md]: Done.

File: src/od/od.rs (hunk 6 above)
Comment (keunwoo) [inline src/od/od.rs]: Done.

File: src/od/od.rs (hunk 5 above)
Comment (keunwoo) [inline src/od/od.rs]: Done.

File: src/od/od.rs (hunk 4 above)
Comment (keunwoo) [inline src/od/od.rs]: Done.

File: src/od/od.rs (hunk 3 above)
Comment (keunwoo) [inline src/od/od.rs]: Done.

File: src/od/od.rs (hunk 2 above)
Comment (keunwoo) [inline src/od/od.rs]: Done.

File: src/od/od.rs (hunk 1 above)
Comment (keunwoo) [inline src/od/od.rs]: Done.

**Comment by keunwoo** [comment]
PTAL

**Comment by Arcterus** [comment]
Thanks! :)

**Pull request merged**